feat: support environments domains for exdns controller

joostvdg opened the pull request:
the exdns controller can take multiple domain filters as an argument.
if a user wants to define unique domain names for different environments, they need to be explicitly configured as a separate domain filter.

Signed-off-by: Joost van der Griendt <joostvdg@gmail.com>

joostvdg committed:
process environment domains for exdns controller
the exdns controller can take multiple domain filters as an argument.
if a user wants to define unique domain names for different environments, they need to be explicitly configured as a separate domain filter.

Signed-off-by: Joost van der Griendt <joostvdg@gmail.com>

joostvdg commented:
/assign @ccojocar 
As discussed last week.

ccojocar commented:
/ok-to-test

ccojocar commented on systems/external-dns/values.tmpl.yaml:
```diff
@@ -11,4 +11,11 @@ external-dns:
+# add default ingress domain  
   - "{{ .Requirements.ingress.domain }}"
+# loop over environments' ingresses and add each domain
+{{- range .Requirements.environments }}
```
I think the main `Requirements.ingress.domain` and `.Requirements.evnvironments.dev.ingress.domain` are duplicated. 

We can remove the `Requirements.ingress.domain` from the list since is the same with `dev` environment ingress.

joostvdg replied:
I wasn't sure. If I leave everything blank in the environments block, and only fill in the root ingress, will jx boot ensure the dev environment's ingress definition gets populated?

ccojocar replied:
Yes. Actually if you check the `jx-requirements.yaml` file right before that stage, it should be populated. It will just copy the values from main ingress into dev ingress.

joostvdg committed:
assuming jx boot fills dev ingress we can skip default
If jx boot ensures the dev environment's ingress domain gets a value, we can rely on that to be there

Signed-off-by: Joost van der Griendt <joostvdg@gmail.com>

ccojocar commented:
@jasongerard Please could you rebase your change and fix the review comment? Thanks!

ccojocar commented:
Closing this in favour of https://github.com/jenkins-x/jenkins-x-boot-config/pull/129
